include/realizations/catchment/Formulation_Manager.hpp

```diff
@@ -128,6 +129,53 @@ namespace realization {
                  */ 
                 this->Simulation_Time_Object = std::make_shared<Simulation_Time>(simulation_time_config);
 
+                /**
+                 * Read the layer descriptions
+                */
+
+                // try to get the json node
+                auto layers_json_array = tree.get_child_optional("layers");
+
+                // check to see if the node existed
+                if (!layers_json_array) {
+                    // If there is no layers section create data for a default surface hydrology
+                    // TODO
+                }
+                else
+                {
+                    for (std::pair<std::string, boost::property_tree::ptree> layer_config : *layers_json_array) 
+                    {
+                        // layer description struct
+                        ngen::LayerDescription layer_desc;
+
+                        // extract and store layer data from the json
+                        layer_desc.name = layer_config.second.get<std::string>("name");
+                        layer_desc.id = layer_config.second.get<int>("id");
+                        layer_desc.time_step = layer_config.second.get<double>("time_step");
+                        boost::optional<std::string> layer_units = layer_config.second.get_optional<std::string>("time_step_units");
+                        if (*layer_units == "") layer_units = "s";
```

Comment:
Need to check if the `optional` is valid to dereference first, and also set the default if it is `boost::none`
```c++
if ( (layer_units && *layer_units == "") || !layer_units ) layer_units = "s";
```

Reply:
This is a bug that needs to get fixed before we merge!

Reply:
This likely works cause `get_optional<std::string>` will return an "empty object" of type `std::string` if the key doesn't exist...so `layer_units` is guaranteed to be initialized to a string, and the `""` if not key is present...there may still be some unthought of edge case that would make `layer_units` uninitialized...